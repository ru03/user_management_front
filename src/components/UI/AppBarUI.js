import React from 'react';
import { AppBar, Button, Toolbar } from '@material-ui/core';
import { Link, useHistory } from 'react-router-dom';
import styles from './AppBarUI.module.css';
import { useAuthDataContext } from '../../providers/authProvider';

const AppBarUI = ({ color = 'default', position = 'fixed' }) => {
  const history = useHistory();
  const { isAuth, onLogout } = useAuthDataContext();

  const onSignOut = () => {
    localStorage.removeItem('token');
    onLogout();
    history.push('/login');
  }

  return (
    <AppBar
      color={color}
      position={position}
    >
      <Toolbar>
        {/* <Typography variant="h6">
        User Management
      </Typography> */}
        <Button color="inherit">
          <Link to='/users' className={styles.links}>Home</Link>
        </Button>
        {
          isAuth
            ? (
              <>
                <Button color="inherit">
                  <Link to='/user' className={styles.links}>Create User</Link>
                </Button>
                <div className={styles.signOut}>
                  <Button color="inherit" onClick={onSignOut}>Sign out</Button>
                </div>
              </>
            )
            : (
              <div className={styles.signOut}>
                <Button color="inherit">
                  <Link to='/login' className={styles.links}>Sign In</Link>
                </Button>
              </div>
            )
        }
      </Toolbar>
    </AppBar>
  );
}

export default AppBarUI;


