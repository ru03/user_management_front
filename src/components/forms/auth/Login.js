import React from 'react';
import { Box, Button, Card, CardContent, CardActions, CardHeader, Grid, makeStyles, TextField } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  header: {
    textAlign: 'center',
    spacing: 10,
  },
  card: {
    width: '30%',
  }
}));
const Login = ({ errors, handleChange, handleSubmit, isSubmitting, touched, values }) => {
  const classes = useStyles();
  return (
    <form onSubmit={handleSubmit} data-testid="loginForm">
      <Box mt="25%">
        <Grid
          container
          direction="column"
          justify="center"
          alignItems="center"
        >
          <Card className={classes.card}>
            <CardHeader title="Login" className={classes.header} />
            <CardContent>
              <Grid lg={12} item>
                <TextField
                  id="email"
                  name="email"
                  label="Email"
                  placeholder="email"
                  fullWidth
                  onChange={handleChange}
                  value={values.email}
                  error={touched.email && errors.email ? true : false}
                  helperText={touched.email && errors.email}
                />
              </Grid>
              <Grid lg={12} item>
                <TextField
                  id="password"
                  name="password"
                  label="Password"
                  type="password"
                  placeholder="password"
                  fullWidth
                  onChange={handleChange}
                  value={values.password}
                  error={touched.password && errors.password ? true : false}
                  helperText={touched.password && errors.password}
                />
              </Grid>
            </CardContent>
            <CardActions>
              <Grid lg={12} item>
                <Button
                  id="login_button"
                  data-testid="buttonSubmit"
                  color="primary"
                  disabled={isSubmitting}
                  variant="contained"
                  type="submit"
                  fullWidth
                >
                  Login
                </Button>
              </Grid>
            </CardActions>
          </Card>
        </Grid>
      </Box>
    </form>
  )
}

export default Login;