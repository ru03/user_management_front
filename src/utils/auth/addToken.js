const getBearerToken = () => {
  const token = localStorage.getItem('token');
  return `Bearer ${token}`;
}

export default getBearerToken;