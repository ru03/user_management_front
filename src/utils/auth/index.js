export { default as addBearerToken } from './addToken';