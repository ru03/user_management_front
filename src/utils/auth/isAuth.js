import jwtDecode from 'jwt-decode';

const isAuthorize = () => {
  const token = localStorage.getItem('token');
  if (!token) return false;
  const { exp } = jwtDecode(token);
  if (new Date() < new Date(exp * 1000)) {
    return true;
  } else {
    localStorage.removeItem('token');
    return false;
  }
}

export default isAuthorize;