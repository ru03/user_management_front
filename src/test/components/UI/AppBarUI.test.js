import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'
import AppBarUI from '../../../components/UI/AppBarUI';
import AuthProvider from '../../../providers/authProvider';
import * as isAuthorize from '../../../utils/auth/isAuth';

describe('FloatButton Component', () => {

  it('mounts', () => {
    const { asFragment } = render(<AuthProvider><AppBarUI /></AuthProvider>, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Is not authorize', () => {
    it('has not authorize buttons', () => {
      const { getByText } = render(<AuthProvider><AppBarUI /></AuthProvider>, { wrapper: MemoryRouter });
      expect(getByText('Home')).toHaveTextContent('Home');
      expect(getByText('Sign In')).toHaveTextContent('Sign In');
    });
  });

  describe('Is authorize', () => {
    it('has authorize buttons', () => {
      jest.spyOn(isAuthorize, 'default').mockReturnValue(true);
      const { getByText } = render(<AuthProvider><AppBarUI /></AuthProvider>, { wrapper: MemoryRouter });
      expect(getByText('Home')).toHaveTextContent('Home');
      expect(getByText('Sign out')).toHaveTextContent('Sign out');
      expect(getByText('Create User')).toHaveTextContent('Create User');
    });
  });
});
