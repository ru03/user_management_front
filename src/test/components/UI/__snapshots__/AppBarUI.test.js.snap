// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatButton Component mounts 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorDefault mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <a
            class="links"
            href="/users"
          >
            Home
          </a>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="signOut"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              class="links"
              href="/login"
            >
              Sign In
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
